### PR TITLE
Fixes #903: Fix autodiscovery endless loop

### LINKED ISSUE #903: Feed detection and redirection endless loop
When trying to find a feed url, Liferea can get into an endless loop, and the UI doesn't really show what is happening. The main hint is that Liferea is using a lot of bandwith while not showing anything in the new feed.
Here is an example of a problematic site : https://www.groupe-ldlc.com/communiques-de-presse/
The page advertises with `<link rel="alternate" type="application/rss+xml" />` a feed url that doesn't really exist but instead of throwing a 404 redirect to https://www.groupe-ldlc.com/, which advertises a feed that isn't there and redirects ...
In fact, there are three feed urls given and only the last one is an actual feed.

So the fault is partly theirs for giving bad links, but it would be nice if Liferea didn't fall into a loop. There shouldn't be a big number of attempts before finding the feed in normal cases, so perhaps we could count those and limit to a fixed number like 3 or 4 ...

### PR DESCRIPTION
**Pull request by lwindolf**
- Add audodiscovery redirect counter to subscription
- Stops after 5 redirects
- Drop mutation of ctxt->failed use return codes instead

### CODE REVIEW

File: src/feed_parser.c [hunk 1]
```diff
@@ -128,18 +130,20 @@ feed_parser_auto_discover (feedParserCtxtPtr ctxt)
...
-		debug0 (DEBUG_UPDATE, "No feed link found!");
-		g_string_append (ctxt->feed->parseErrors, _("The URL you want Liferea to subscribe to points to a webpage and the auto discovery found no feeds on this page. Maybe this webpage just does not support feed auto discovery."));
+
+		return TRUE;
 	}
+
+	debug0 (DEBUG_UPDATE, "No feed link found!");
+	g_string_append (ctxt->feed->parseErrors, _("The URL you want Liferea to subscribe to points to a webpage and the auto discovery found no feeds on this page. Maybe this webpage just does not support feed auto discovery."));
+	return FALSE;
 }
```
Comment (Leiaz) [inline src/feed_parser.c:147]: I had not noticed the implicit return here before, in the sea of mostly useless warnings. Hopefully the compiler would have complained louder if we had tried to use the value.

File: src/feed_parser.c [hunk 2]
```diff
@@ -174,15 +177,13 @@ feed_parse (feedParserCtxtPtr ctxt)
 {
 	xmlNodePtr	xmlNode = NULL, htmlNode = NULL;
 	xmlDocPtr	xmlDoc, htmlDoc;
-	gboolean	success = FALSE;
+	gboolean	autoDiscovery = FALSE, success = FALSE;
 	gboolean	htmlParsing = FALSE;
```
Comment (Leiaz) [inline src/feed_parser.c:181]: On the topic of warnings, the `htmlParsing` variable is unused.

**Review by Leiaz — APPROVED** [review]

File: src/feed_parser.c (hunk 2 above)
Comment (lwindolf) [reply src/feed_parser.c]: Fixed

File: src/feed_parser.c (hunk 1 above)
Comment (lwindolf) [reply src/feed_parser.c:147]: True the warnings are a hassle. I'm trying to reduce them more and more.

**Pull request merged**